crates/sui-core/src/transaction_manager.rs

```diff
@@ -458,7 +458,13 @@ impl TransactionManager {
                 }
 
                 for key in input_object_keys.iter() {
-                    object_availability.insert(*key, None);
+                    if key.is_cancelled() {
+                        // Cancelled txn objects should always be available immediately.
+                        // Don't need to wait on these objects for execution.
+                        object_availability.insert(*key, Some(true));
```

Comment:
In that case, why do you need to insert it?

Reply:
The `input_object_keys` when creating `PendingCertificate` below are using all the input keys. Later, it uses `object_availability` to check if they are available. So we need to record shared objects availability here as well. I guess I could also make `input_object_keys` to only contain non-shared object, but I tried not to change too much code here.